### ISSUE TITLE
Matrix Matrix exponntial 

### ISSUE THREAD

**Issue by abhinav28071999** [sympy/sympy#18806]
Sympy probably doesn't support this https://en.wikipedia.org/wiki/Matrix_exponential#Matrix-matrix_exponentials
It should be easy to implement. Only question is that should this be a  different function or can be included in  `pow` function.

**Comment by sylee957** [comment]
I don't think that it's easy to fit such in design goals of `pow`, because of the existance of left and right exponential.
So I'd rather expect to use a separate functions for matrix-matrix exponentials, because I don't think that it should be able to make the original `pow` more convenient.

**Comment by jainsee24** [comment]
can't we simply use scipy.linalg.expm for matrix exponential?

**Comment by sylee957** [comment]
Is Pade approximation symbolic?
I would expect non-numeric methods for this. And better use `mpmath.expm` than `scipy` if we are going to wrap the numeric methods with evalf logic.

**Comment by czgdp1807** [comment]
Are there any use cases where we need matrix-matrix exponential instead of calling, `(Y.log()*X).exp()` for left and something similar for right exponential?
